src/EnergyPlus/CrossVentMgr.cc

```diff
@@ -166,181 +164,168 @@ namespace CrossVentMgr {
         state.dataCrossVentMgr->HA_R = 0.0;
 
         // Is the air flow model for this zone set to UCSDCV Cross Ventilation?
-        if (state.dataRoomAirMod->IsZoneCV(ZoneNum)) {
+        if (state.dataRoomAir->IsZoneCrossVent(ZoneNum)) {
+
+            Real64 zoneJetRecAreaRatio = state.dataRoomAir->JetRecAreaRatio(ZoneNum);
+                
             // WALL Hc, HA and HAT calculation
-            for (int Ctd = state.dataUCSDShared->PosZ_Wall((ZoneNum - 1) * 2 + 1); Ctd <= state.dataUCSDShared->PosZ_Wall((ZoneNum - 1) * 2 + 2);
-                 ++Ctd) {
-                int SurfNum = state.dataUCSDShared->APos_Wall(Ctd);
+            for (int Ctd = state.dataRoomAir->PosZ_Wall(ZoneNum).beg; Ctd <= state.dataRoomAir->PosZ_Wall(ZoneNum).end; ++Ctd) {
+                int SurfNum = state.dataRoomAir->APos_Wall(Ctd);
+                if (SurfNum == 0) continue;
+
+                auto const &surf = state.dataSurface->Surface(SurfNum);
                 state.dataSurface->SurfTAirRef(SurfNum) = DataSurfaces::RefAirTemp::AdjacentAirTemp;
```

Comment:
This function is called from the CalcUCSDCV (now CalcCrossVent) function. Do we need to set this (line 177) enum each iteration? This will not change for a specific surface will it?

    SurfaceConvectionAlgorithm:Inside:UserCurve,
      A2 , \field Reference Temperature for Convection Heat Transfer
           \note Controls which temperature is differenced from surface temperature when using the Hc value
           \type choice
           \key MeanAirTemperature
           \key AdjacentAirTemperature
           \key SupplyAirTemperature

Reply:
I don't know.